### PR TITLE
Add timestamp for debug logs

### PR DESCRIPTION
**Pull request by xm1k3**
## Proposed changes

Added timestamp on --debug

```
[DBG] [2023-01-01T22:48:58+01:00] ...
```

## Changes Made
This PR includes the following changes:
- Added the `SetTimestamp` method to the `Logger` struct in [gologger](https://github.com/projectdiscovery/gologger), which allows the user to enable or disable automatic timestamps for log messages, and to set the minimum level of log messages for which timestamps should be included.
- Updated the `init()` function to set the default maximum log level to `LevelInfo`, and added a timestamp option for the formatter and writer.

## How to use
The package provides a `DefaultLogger` variable that can be used to log messages with different levels (`Debug`, `Info`, `Error`, etc.). You can configure the behavior of the `DefaultLogger` using various methods, such as `SetTimestamp`.

An example of how to use this package:
```go
package main

import (
	"github.com/projectdiscovery/gologger"
)

func main() {
    gologger.DefaultLogger.SetTimestamp(true, levels.LevelDebug)
    gologger.Debug("This message will include a timestamp because its level is greater than or equal to LevelDebug")
}
```

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [x] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
This approach doesn't seem very scalable. Each new added instruction will require the `.Timestamp()` call. I think we should either define an helper function that will invoke:

```go
gologger.Debug().Timestamp().Msgf("xxx)
```

or make this configurable via gologger itself, for example with global setter like:

```go
gologger.EnableTimestamp()
gologger.EnableTimestampForLevel(Debug|Info|etc)
```

**Comment by xm1k3** [comment]
Important note: 
Before merging this PR, make sure that the PR (https://github.com/projectdiscovery/gologger/issues/29) on the Logger has also been merged. The function called in this PR currently does not exist on the master branch of the Logger and would cause issues.

File: v2/cmd/nuclei/main.go [hunk 1]
```diff
@@ -322,6 +322,9 @@ on extensive configurability, massive extensibility and ease of use.`)
 
 	_ = flagSet.Parse()
 
+	logger := gologger.Logger{}
+	logger.SetTimestamp(options.Timestamp)
```
Comment (Mzack9999) [inline v2/cmd/nuclei/main.go:326]:
```
logger.SetTimestamp(options.Timestamp, levels.LevelDebug)
```

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
You can test with the specific gologger commit with:
```console
go get -v github.com/projectdiscovery/gologger@1c357ca
```

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
You might need to point gologger with the correct reference:
```
go get -v github.com/projectdiscovery/gologger@xxx
```

**Review by Mzack9999 — APPROVED** [review]

**Review by ehsandeep — CHANGES_REQUESTED** [review]
- [ ] failing test
- [ ] update PR with a working example.

**Pull request merged**